include/ppx/geometry.h

```diff
@@ -157,43 +167,68 @@ class Geometry
         char*       GetData() { return DataPtr(mData); }
         const char* GetData() const { return DataPtr(mData); }
         uint32_t    GetElementCount() const;
+        uint32_t    GetDataSize() const;
 
         // Trusts that calling code is well behaved :)
         //
+        // Write(count, pValues...) is used to write a chunk of data at specified offset
+        // If called with offset=-1, will write at the end and adjust mOffset accordingly
         template <typename T>
-        void Append(const T& value)
+        void Write(uint32_t count, const T* pValues, int32_t offset = -1)
```

Comment:
A few thoughts:
This breaks the pattern and uses  in the comment above the class. 
At the minimum the class comments need to be updated.
But stepping back for a second, can you accomplish the same thing with the Use #1 with just setting data and managing it directly. The comment above specifically says "Mixing them will most likely lead to undefined behavior." and I think this new function specifically combines two uses?